metrics/calculate_PSNR_SSIM.py

```diff
@@ -1,131 +1,127 @@
-'''
-calculate the PSNR and SSIM.
-same as MATLAB's results
-'''
-import os
+import glob
 import math
-import numpy as np
+import os
+
 import cv2
-import glob
+import numpy as np
 
 
 def main():
-    # Configurations
-
-    # GT - Ground-truth;
-    # Gen: Generated / Restored / Recovered images
-    folder_GT = '/mnt/SSD/xtwang/BasicSR_datasets/val_set5/Set5'
-    folder_Gen = '/home/xtwang/Projects/BasicSR/results/RRDB_PSNR_x4/set5'
-
+    """
+    Calculate the PSNR and SSIM. Same as MATLAB's results.
+
+    Terms:
+        GT: Ground-truth
+        Gen: Generated/Restored/Recovered images
+    """
+    # Config
+    folder_gt = '/mnt/SSD/BasicSR_datasets/val_set5/Set5'
+    folder_gen = '/home/Projects/BasicSR/results/RRDB_PSNR_x4/set5'
     crop_border = 4
     suffix = ''  # suffix for Gen images
-    test_Y = False  # True: test Y channel only; False: test RGB channels
-
-    PSNR_all = []
-    SSIM_all = []
-    img_list = sorted(glob.glob(folder_GT + '/*'))
-
-    if test_Y:
-        print('Testing Y channel.')
-    else:
-        print('Testing RGB channels.')
+    test_y = False  # True: test Y channel only, will convert to YCbCr color space; False: test RGB channels
 
-    for i, img_path in enumerate(img_list):
+    print('Testing [%s] channels.' % ('Y' if test_y else 'R,G,B'))
+    psnr_all = []
+    ssim_all = []
+    for i, img_path in enumerate(sorted(glob.glob(folder_gt + '/*'))):
         base_name = os.path.splitext(os.path.basename(img_path))[0]
-        im_GT = cv2.imread(img_path) / 255.
-        im_Gen = cv2.imread(os.path.join(folder_Gen, base_name + suffix + '.png')) / 255.
+        im_gt = cv2.imread(img_path) / 255.
+        im_gen = cv2.imread(os.path.join(folder_gen, base_name + suffix + '.png')) / 255.
 
-        if test_Y and im_GT.shape[2] == 3:  # evaluate on Y channel in YCbCr color space
-            im_GT_in = bgr2ycbcr(im_GT)
-            im_Gen_in = bgr2ycbcr(im_Gen)
-        else:
-            im_GT_in = im_GT
-            im_Gen_in = im_Gen
+        if test_y and im_gt.shape[2] == 3:
+            # evaluate on Y channel in YCbCr color space
+            # TODO: Why? Is there some kind of benefit I'm not seeing here?
```

Comment:
The original PSNR formulation, the formula in Matlab and most CV papers only calculate PSNR on the Y channel. It's the standard.